easygcm/src/main/java/eu/inloop/easygcm/GcmHelper.java

```diff
@@ -29,9 +26,11 @@
     private GcmListener mGcmListener;
     private GcmServicesHandler mCheckServicesHandler;
     static volatile boolean sLoggingEnabled = true;
+
     private final AtomicBoolean mRegistrationRunning = new AtomicBoolean(false);
-    private static final int DEFAULT_BACKOFF_MS = 2000;
-    private static final int MAX_RETRIES = 5;
+    private RegistrationListener mRegistrationListener;
+
+    private Set<String> mSenderIdsToRegister = new HashSet<>();
```

Comment:
It's not enough to save ids in the memory for connectivity change. The connectivity change can happen after the app is killed. It should be persisted.